app/src/main/java/aa/android/activities/SettingsActivity.kt

```diff
@@ -65,6 +72,54 @@ class SettingsActivity : BaseActivity() {
         val musicButton = findViewById<ImageView>(R.id.music_button)
         val closeCallsButton = findViewById<ImageView>(R.id.close_calls_button)
         val languageButton = findViewById<ImageView>(R.id.language_button)
+        val languages = ArrayList<String>()
+        languages.addAll(getResources().getStringArray(R.array.languages))
+        languages.add(0,resources.getString(R.string.setting_language))
+        val spinner = findViewById<Spinner>(R.id.language_spinner)
+
+        val adapter = object : ArrayAdapter<String>(this, R.layout.my_spinner_style, languages) {
+            override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+                val view = super.getDropDownView(position, convertView, parent)
+                if (position == 0) {
+                    view.visibility = View.GONE
+                } else {
+                    view.visibility = View.VISIBLE
+                }
+                return view
+            }
+
+            override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+                return super.getView(position, convertView, parent)
+                val v = super.getDropDownView(position, convertView, parent)
+                (v as TextView).setGravity(Gravity.CENTER)
+                return v
+            }
+        }
+        adapter.setDropDownViewResource(R.layout.spinner_dropdown_item)
+        spinner.adapter = adapter
+        spinner.setSelection(0)
+        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                if (position == 1) {
```

Comment:
Is it possible to implement this part without using if?
e.g. use an array, use the index and then choose item at that index, so that adding more languages doesn't require adding more conditions?